### ISSUE TITLE
Info/Hint Settings swapped 

### ISSUE THREAD

**Issue by DRKSLV** [rust-analyzer/rust-analyzer#8485]
![image](https://user-images.githubusercontent.com/56208328/114429965-329a5a00-9bbe-11eb-867d-4a04ff6e5207.png)

If i register dead_code under the Setting "Warning as Info", that claims to "display warning with HINT severity" and to simply fade the text color, it underlines them with blue squiggly lines

If i register dead_code under the Setting "Warning as Hint", that claims to "displayed warnings with INFO severity" and to underline text with blue squiggly lines, it fades the text color

So basically the Settings have contradictary descriptions and swapped.
This is kinda confusing, that's all, the actual functionality seems to be working

(v0.2.554, VsCode Windows 64bit)
